src/com/amazon/corretto/crypto/provider/RsaCipher.java

```diff
@@ -211,25 +234,119 @@ protected AlgorithmParameters engineGetParameters() {
     }
 
     @Override
-    protected void engineInit(final int opmode, final Key key, final SecureRandom random)
-            throws InvalidKeyException {
+    protected void engineInit(final int opmode, final Key key, final AlgorithmParameterSpec params,
+            final SecureRandom random)
+                    throws InvalidKeyException, InvalidAlgorithmParameterException {
+        if (params != null) {
+            if (params instanceof OAEPParameterSpec) {
+                final OAEPParameterSpec oaep = (OAEPParameterSpec) params;
+                if (!"MGF1".equalsIgnoreCase(oaep.getMGFAlgorithm()) ||
+                        oaep.getMGFParameters() == null ||
+                        !(oaep.getMGFParameters() instanceof MGF1ParameterSpec)) {
+                    throw new InvalidAlgorithmParameterException();
+                }
+                final PSource pDefault = PSource.PSpecified.DEFAULT;
+                final PSource psrc = oaep.getPSource();
+                if (psrc == null || !pDefault.getAlgorithm().equalsIgnoreCase(psrc.getAlgorithm())) {
+                    throw new InvalidAlgorithmParameterException();
+                }
+                // Only support empty label
```

Comment:
hm, i'd generally think of that as a customer-driven feature request rather than a maintenance TODO, but i can add a TODO comment.

Reply:
I'm not really picky about how we track it. I just want to keep track that this was a choice rather than a technical limitation. Previously we only supported empty labels due to a technical limitation and so we *couldn't* add it. Now we could with minimal work, so we should make a note that this is an option if needed later on. Exact wording and mechanism is less important to me.

Reply:
ah, shoot. i forgot to add a comment/TODO here in the most recent revision. i'll fix that either on the next revision of this PR (if needed) or in the next PR.